serde-reflection/src/trace.rs

```diff
@@ -285,6 +288,60 @@ impl Tracer {
         self.record_container(samples, name, format, value, false)
     }
 
+    /// Removes aliases for a field in a container from the registry.
+    ///
+    /// The serde doesn't differentiate between alias names from field names so
+    /// this is a way to inform the registry of aliases of a particular field in a
+    /// container.
+    pub fn ignore_aliases(
+        &mut self,
+        container_name: &'static str,
+        aliases: &[&'static str],
+    ) -> Result<()> {
+        let container_format = match self.registry.get_mut(container_name) {
+            None => {
+                return Err(Error::Custom(format!(
+                    "cannot alias. container not found: {}",
+                    container_name
+                )))
+            }
+            Some(container_format) => container_format,
+        };
+        match container_format {
+            ContainerFormat::Struct(named_formats) => {
+                let (non_alias_formats, alias_formats): (Vec<_>, Vec<_>) = named_formats
+                    .iter()
+                    .cloned()
+                    .partition(|format| !aliases.contains(&format.name.deref()));
+
+                if alias_formats.len() != aliases.len() {
+                    return Err(Error::Custom(format!(
+                        "cannot ignore alias. not all aliases are found in container: {:?}",
```

Comment:
So if the user says "ignore alias x" but x isn't found, this will throw an error, is that right?

Reply:
Yes, this was intentional.